### PR TITLE
chore(debug): migrate transparent setup tasks to v3

### PR DESCRIPTION
**Pull request by kuojianlu**

### CODE REVIEW

File: packages/fx-core/templates/core/v3Migration/js.ts.app.local.yml [hunk 1]
```diff
@@ -17,6 +90,72 @@ deploy:
...
+    with:
+      target: ./bot/.env.teamsfx.local
+      envs:
+        M365_CLIENT_ID: $\{{AAD_APP_CLIENT_ID}}
+        M365_CLIENT_SECRET: $\{{SECRET_AAD_APP_CLIENT_SECRET}}
+        M365_TENANT_ID: $\{{AAD_APP_TENANT_ID}}
+        M365_AUTHORITY_HOST: $\{{AAD_APP_OAUTH_AUTHORITY_HOST}}
+        INITIATE_LOGIN_ENDPOINT: $\{{PROVISIONOUTPUT__AZUREWEBAPPBOTOUTPUT__SITEENDPOINT}}/auth-start.html
+        {{#if ssoTab}}
+        M365_APPLICATION_ID_URI: api://$\{{PROVISIONOUTPUT__AZURESTORAGETABOUTPUT__DOMAIN}}/botid-$\{{BOT_ID}}
```
Comment (XiaofuHuang) [inline packages/fx-core/templates/core/v3Migration/js.ts.app.local.yml:138]: The `M365_APPLICATION_ID_URI` is different according project type. See `generateAppIdUri` in `packages\fx-core\src\core\middleware\utils\v3MigrationUtils.ts`

Comment (kuojianlu) [reply packages/fx-core/templates/core/v3Migration/js.ts.app.local.yml:138]: Yes, so I use the if else syntax in this yml template.

File: packages/fx-core/src/core/middleware/utils/debug/appLocalYmlGenerator.ts [hunk 2]
```diff
@@ -1,16 +1,43 @@
...
 // Licensed under the MIT license.
 
-import { ProjectSettings } from "@microsoft/teamsfx-api";
+import { AzureSolutionSettings, ProjectSettings } from "@microsoft/teamsfx-api";
 import { BuildArgs } from "../../../../component/driver/interface/buildAndDeployArgs";
 import { InstallToolArgs } from "../../../../component/driver/tools/interfaces/InstallToolArgs";
 import { BaseAppYmlGenerator } from "../appYmlGenerator";
 
 export class AppLocalYmlConfig {
+  registerApp?: {
```
Comment (XiaofuHuang) [inline packages/fx-core/src/core/middleware/utils/debug/appLocalYmlGenerator.ts:10]: What about organize attributes according to project type?
```
{
    tab?: {
        sso?: boolean;
        domain?: string;
        endpoint?: string;
        port?: number;
    },
    bot?: {
          ......
    }
}
```

File: packages/fx-core/templates/core/v3Migration/js.ts.app.local.yml [hunk 3]
```diff
@@ -1,5 +1,78 @@
...
+  {{#if bot}}
+  - uses: botAadApp/create
+    with:
+      name: $\{{CONFIG__MANIFEST__APPNAME__SHORT}}-bot
+
+  - uses: botFramework/create
+    with:
+      botId: $\{{BOT_ID}}
+      name: $\{{CONFIG__MANIFEST__APPNAME__SHORT}}-bot
+      messagingEndpoint: $\{{PROVISIONOUTPUT__AZUREWEBAPPBOTOUTPUT__SITEENDPOINT}}/api/messages
```
Comment (XiaofuHuang) [inline packages/fx-core/templates/core/v3Migration/js.ts.app.local.yml:30]: Use `namingConverterV3` to generate the placeholder.

File: packages/fx-core/src/core/middleware/utils/debug/appLocalYmlGenerator.ts (hunk 2 above)
Comment (kuojianlu) [reply packages/fx-core/src/core/middleware/utils/debug/appLocalYmlGenerator.ts:10]: If so, the syntax in yml template will be complex. For example, to check if `aadApp/create` is needed, we should check `tab.sso` and `bot.sso`.

File: packages/fx-core/templates/core/v3Migration/js.ts.app.local.yml (hunk 3 above)
Comment (kuojianlu) [reply packages/fx-core/templates/core/v3Migration/js.ts.app.local.yml:30]: Good catch! As it depends on bicep content, let me use it in another PR.

**Review by swatDong — APPROVED** [review]

**Review by XiaofuHuang — DISMISSED** [review]

**Review by swatDong — APPROVED** [review]

**Review by 1yefuwang1 — APPROVED** [review]

**Pull request merged**